src/Hl7.Fhir.Base/Serialization/engine/PocoSerializationEngine_Json.cs

```diff
@@ -0,0 +1,68 @@
+#nullable enable
+
+using Hl7.Fhir.Introspection;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Utility;
+using System;
+using System.Linq;
+using System.Text.Json;
+
+namespace Hl7.Fhir.Serialization;
+
+internal partial class PocoSerializationEngine
+{
+    private readonly FhirJsonPocoDeserializerSettings? _jsonDeserializerSettings;
+    private readonly FhirJsonPocoSerializerSettings? _jsonSerializerSettings;
+    
+    private BaseFhirJsonPocoDeserializer? _jsonDeserializer;
+    private BaseFhirJsonPocoSerializer? _jsonSerializer;
+    
+    private BaseFhirJsonPocoDeserializer getJsonDeserializer() => 
+        _jsonDeserializer ??= new BaseFhirJsonPocoDeserializer(_inspector, _jsonDeserializerSettings!);
+
+    private BaseFhirJsonPocoSerializer getJsonSerializer() =>
+        _jsonSerializer ??= new BaseFhirJsonPocoSerializer(_inspector.FhirRelease, _jsonSerializerSettings!);
+    
+    /// <inheritdoc />
+    public Resource DeserializeFromJson(string data)
+    {
+        return (Resource)deserializeAndFilterErrors(() =>
+        { 
+            _ = getJsonDeserializer().TryDeserializeResource(data, out var instance, out var issues);
```

Comment:
What happens when this fails, we still return the resource, which could be null?
No, it can't be, because Rider/VS would throw an error.

Reply:
I more or less copied this from the previous implementation, deserializeAndFilterErrors either returns an instance or throws a DFE (where the instance is nullable).

We should check if the ostrich parser breaks this behaviour with completely malformed json by returning null and not reporting any issues